fix(forms): remove max default on form stepper

jefflembeck opened the pull request:
There should be no max default for the stepper component

BREAKING CHANGE: This could break tests down the line and any
assumptions made when the max value was not set. Originally, the max
defaulted to 99999. If a user had left the `max` attribute out of the
input element, with this change, there is no longer that default max to
fall back on

jefflembeck committed:
fix(forms): remove max default on form stepper

There should be no max default for the stepper component

BREAKING CHANGE: This could break tests down the line and any
assumptions made when the max value was not set. Originally, the max
defaulted to 99999. If a user had left the `max` attribute out of the
input element, with this change, there is no longer that default max to
fall back on

jefflembeck committed:
chore(forms): Bump version because of merge issue

aredridel commented on src/pivotal-ui/components/forms/stepper.js:
```diff
@@ -24,7 +24,7 @@ FormStepper.prototype.set = function(number){
   if( number <= this.min ){
     number = this.min;
-  } else if (number >= this.max ){
+  } else if (this.max && number >= this.max ){
```
Maybe `this.max != null` instead of the truthiness check?

jefflembeck replied:
That's a solid idea. There's the shot that `max="0"` happens.

jefflembeck replied:
Ooo.. however, in the constructor, an undefined max would lead to `NaN` being the value here. I can check if `isNaN`

aredridel replied:
Fair! Or change the constructor to squash `NaN` to `null`.

aredridel commented:
In the merge message, maybe we can explain more about what the user of this module has to change, rather than what it could break.

> BREAKING CHANGE: the max attribute must be supplied if a max is wanted. The default has changed from `99999` to no max enforced.

maybe?

jefflembeck committed:
fix(forms): be more explicit about this.max

this.max should be set to null if it doesn't exist or is nonsense for
more explicit comparisons later

aredridel commented:
Looks good to me! Nice changes!

aredridel commented:
:sheep: it!
